API/Repositories/Implementations/ApiMatchRatingStatsRepository.cs

```diff
@@ -3,85 +3,68 @@
-        //     .Where(ra =>
-        //         ra.PlayerId == playerId
-        //         && ra.Ruleset == (Ruleset)ruleset
-        //         && ra.Timestamp >= dateMin
-        //         && ra.Timestamp <= dateMax
-        //     )
-        //     .Select(ra => new
-        //     {
-        //         ra.Timestamp, // Use Timestamp for grouping as fallback
-        //         DataPoint = new PlayerRatingChartDataPointDTO
-        //         {
-        //             Name = ra.RatingAdjustmentType == 0 ? "Decay" : "Adjustment",
-        //             RatingBefore = ra.RatingBefore,
-        //             RatingAfter = ra.RatingAfter,
-        //             VolatilityBefore = ra.VolatilityBefore,
-        //             VolatilityAfter = ra.VolatilityAfter,
-        //             IsAdjustment = true,
-        //             Timestamp = ra.Timestamp
-        //         }
-        //     })
-        //     .ToListAsync();
-        //
-        // // Combine data points, converting Match.StartTime and RatingAdjustment.Timestamp to Date for grouping
-        // var combinedDataPoints = matchRatingStats
-        //     .Select(mrs => new { mrs.StartTime.Date, mrs.DataPoint })
-        //     .Concat(ratingAdjustments.Select(ra => new { ra.Timestamp.Date, ra.DataPoint }))
-        //     .GroupBy(x => x.Date)
-        //     .OrderBy(g => g.Key)
-        //     .Select(g => g.Select(x => x.DataPoint).ToList())
-        //     .ToList();
+        // Default date range to min and max values if not provided
+        dateMin ??= DateTime.MinValue;
+        dateMax ??= DateTime.MaxValue;
+
+        // Fetch Match Rating Stats and group by Match.StartTime.Date
+        var adjustments = await _context
+            .RatingAdjustments
+            .Where(ra =>
+                ra.PlayerId == playerId
+                && ra.AdjustmentType != RatingAdjustmentType.Initial
+                && ra.Match!.Tournament.Ruleset == ruleset
+                && ra.Timestamp >= dateMin
+                && ra.Timestamp <= dateMax
+            )
+            .OrderBy(ra => ra.Timestamp)
+            .Select(ra => new
+            {
+                ra.Timestamp,
+                DataPoint = new PlayerRatingChartDataPointDTO
+                {
+                    Name = ra.Match!.Name ?? "<Unknown>",
+                    MatchId = ra.MatchId,
+                    MatchOsuId = ra.Match.OsuId,
+                    MatchCost = _context.PlayerMatchStats
+                        .Where(pms => pms.PlayerId == ra.PlayerId &&
+                                      pms.MatchId == ra.MatchId)
+                        .Select(pms => pms.MatchCost)
+                        .FirstOrDefault(),
+                    RatingBefore = ra.RatingBefore,
+                    RatingAfter = ra.RatingAfter,
+                    VolatilityBefore = ra.VolatilityBefore,
+                    VolatilityAfter = ra.VolatilityAfter,
+                    IsAdjustment = true,
```

Comment:
Is this a legacy indicator? Why is it always set to `true`?

Reply:
Legacy yeah, this is from before we had `RatingAdjustmentType` I'm pretty sure.